core/src/gl/mesh.cpp

```diff
@@ -34,25 +34,15 @@ MeshBase::MeshBase(std::shared_ptr<VertexLayout> _vertexLayout, GLenum _drawMode
 }
 
 MeshBase::~MeshBase() {
-
-    auto vaos = m_vaos;
-    auto glVertexBuffer = m_glVertexBuffer;
-    auto glIndexBuffer = m_glIndexBuffer;
-
-    m_disposer([=](RenderState& rs) mutable {
-        // Deleting a index/array buffer being used ends up setting up the current vertex/index buffer to 0
-        // after the driver finishes using it, force the render state to be 0 for vertex/index buffer
-        if (glVertexBuffer) {
-            rs.vertexBufferUnset(glVertexBuffer);
-            GL::deleteBuffers(1, &glVertexBuffer);
+    if (m_rs) {
+        if (m_glVertexBuffer) {
+            m_rs->queueBufferDeletion(m_glVertexBuffer);
```

Comment:
Just thought that could lock only once by changing the interface a bit, by queuing vao/index/vertex buffer at the same time, same for shaders.

Reply:
Good idea, I'll add that now.

Reply:
Added! I left VAOs separate from buffers because it was a less obvious change.